### ISSUE TITLE
Accepting received does not work if using `VerifierSettings.DerivePathInfo`

### ISSUE THREAD

**Issue by batzen** [matkoch/resharper-verify#4]
Trying to accept the received results as verified seems to do nothing.

Having seen the changes you made in 2021.2.1 i noticed that this will never work in our case, or in case someone uses a different folder to store the files.
We are using `VerifierSettings.DerivePathInfo` to change the path to not be side-by-side with the test file, but produce a separate directory beside the test file. But the exception/error text produced by verify only contains the filename, not the path.
As the plugin uses that text to extract the filename, it doesn't currently seem to be possible to solve this only on the plugin side.

@SimonCropp Maybe you have an idea on how to solve this? Maybe an option to use the full paths inside `ProcessNotEquals` instead of `Path.GetFileName`? Or maybe append the full paths at the bottom of the result?

Versions used:
Rider: 2021.2.1
Plugin: 2021.2.1

**Comment by SimonCropp** [comment]
@matkoch  i could add the two file paths in the exception so they could be parsed

**Comment by SimonCropp** [comment]
this will be fixed in the next release via https://github.com/matkoch/resharper-verify/commit/137bad320cfaa4e6de9c3ac146be175c31f44ccb

**Comment by batzen** [comment]
Now i can't accept with and without custom paths. @SimonCropp 
The gif is without custom paths:
![verify_accept](https://user-images.githubusercontent.com/1971030/153030622-db020d9e-6fca-4f4e-9bc1-616d81d81778.gif)

**Comment by matkoch** [comment]
can you try running in internal mode? i'd expect an exception to pop up

**Comment by batzen** [comment]
That didn't produce any exception.
The only info in `idea.log` is:
```
2022-02-08 18:19:54,264 | INFO  | RiderAnAction                  | Executing ReSharper action 'UnitTestSession.VerifyAccept'
2022-02-08 18:19:54,264 | INFO  | FrontendActionHandlers         | >> Begin backend 'UnitTestSession.VerifyAccept' action session
2022-02-08 18:19:54,264 | INFO  | FrontendActionHandlers         | << End backend 'UnitTestSession.VerifyAccept' action session
```

**Comment by matkoch** [comment]
are you sure internal mode is enabled?

**Comment by SimonCropp** [comment]
can you share a minimal repro?

**Comment by SimonCropp** [comment]
@batzen did you update to the latest version of Verify?

**Comment by batzen** [comment]
@SimonCropp Repro can be found at https://github.com/batzen/XAMLTools
Just delete `XAMLCombinerTests.TestOutput.verified.xaml` and run the tests.
Versions are all latest (Rider, Plugin and Verify).

**Comment by SimonCropp** [comment]
seems we get an unhandled exception in GetExceptionInfo
```
JetBrains.Diagnostics.Assertion+AssertionException: type != null
   at JetBrains.Diagnostics.Assertion.Fail(String message)
   at JetBrains.Diagnostics.Assertion.NotNull[T](T value, String message)
   at JetBrains.ReSharper.UnitTestFramework.Execution.TestException.Marshaller.Unmarshal(UnsafeReader reader)
   at JetBrains.Application.PersistentMap.LevelDbMap`2.TryGetValue(TKey key, TValue& value)
   at JetBrains.Application.PersistentMap.OptimizedPersistentSortedMap`2.TryGetValueCore(TKey key, TValue& value, Boolean useContainsToOptimizeValueUnmarshal)
   at JetBrains.Application.PersistentMap.OptimizedPersistentSortedMap`2.TryGetValue(TKey key, TValue& value)
   at JetBrains.Application.PersistentMap.OptimizedPersistentSortedMap`2.get_Item(TKey key)
   at JetBrains.ReSharper.UnitTestFramework.Execution.UnitTestResultData.GetExceptionInfo(Int32 pos)
   at Extensions.HasVerifyException(UnitTestResultData result) in C:\Code\VerifyTests\resharper-verify\src\dotnet\ReSharperPlugin.Verify\Extensions.cs:line 11
   at ReSharperPlugin.Verify.VerifyAcceptAction.Update(IDataContext context, ActionPresentation presentation, DelegateUpdate nextUpdate) in C:\Code\VerifyTests\resharper-verify\src\dotnet\ReSharperPlugin.Verify\VerifyAcceptAction.cs:line 50
   at JetBrains.Application.UI.ActionsRevised.Handlers.ExecutableActionEvaluator.Evaluate(IAction handler, EvaluatedAction result, IDataContext context)
   at JetBrains.ReSharper.Feature.Services.Actions.ServicesActionEvaluator.Evaluate(IAction handler, EvaluatedAction result, IDataContext context)
   at JetBrains.Application.UI.ActionsRevised.Handlers.BoundAction.Evaluate(IDataContext context)
   at JetBrains.Application.UI.ActionsRevised.Handlers.ActionHandlers.Evaluate(IActionDefWithId action, IDataContext context)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActions.CreateEvaluatedActionDescriptor(IActionDefWithId actionDefWithId, IDataContext dataContext, IActionHandlers handlers)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActionsAsyncAvailability.UpdateAction(ActionUpdateRequestData req, IDataContext dataContext)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActionsAsyncAvailability.UpdateOnBackgroundThread(IPriorityQueue`1 actionsToUpdate)
```

**Comment by SimonCropp** [comment]
@matkoch side note: shouldnt that bubble up to the user in some way?

**Comment by matkoch** [comment]
only in eap

**Comment by matkoch** [comment]
which version?

**Comment by SimonCropp** [comment]
launching from 
```
JetBrains Rider 2022.1 EAP 5
Build #RD-221.4906.10, built on March 5, 2022
Licensed to JetBrains Rider EAP user: simonc
Expiration date: April 3, 2022
Runtime version: 11.0.14.1+1-b2043.11 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
.NET Framework 4.0.30319.42000
GC: G1 Young Generation, G1 Old Generation
Memory: 4000M
Cores: 16
Registry:
    rdclient.asyncActions=false
    indexing.enable.entity.provider.based.indexing=false
    ide.new.project.model.index.case.sensitivity=true

Non-Bundled Plugins:
    verify-rider (2022.1.0-eap03)

```

and the version that launches is 

```
JetBrains Rider 2022.1 EAP 3
Build #RD-221.4501.83, built on February 17, 2022
Licensed to JetBrains Rider EAP user: simonc
Expiration date: March 19, 2022
Runtime version: 11.0.14+9-b1993.2 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
.NET Framework 4.0.30319.42000
GC: G1 Young Generation, G1 Old Generation
Memory: 1500M
Cores: 16
Registry:
    rdclient.asyncActions=false
    indexing.enable.entity.provider.based.indexing=false
    ide.new.project.model.index.case.sensitivity=true

Non-Bundled Plugins:
    verify-rider (9999.0.0)

```

**Comment by matkoch** [comment]
That should be fixed in eap5, but I guess it's not the original issue here

**Comment by SimonCropp** [comment]
so how do i change the version that rider launched when i go to debug?

**Comment by matkoch** [comment]
Same as here: https://github.com/matkoch/resharper-verify/commit/b90fbeb92f7d0bd39b0fd2f9ee5106816597d4dc

**Comment by SimonCropp** [comment]
so it seems `TestException.Type` has started to return am empty string for the Type. Perhaps since `VerifyException` has no namespace?

![image](https://user-images.githubusercontent.com/122666/157754481-93d83b24-fb6f-43d3-a243-91c13cc53f68.png)

**Comment by SimonCropp** [comment]
@matkoch thoughts on the above?

**Comment by matkoch** [comment]
I haven't seen anything like that. Can you give some pointers when it happens (repro/revision)? Does it still happen on EAP6?

**Comment by SimonCropp** [comment]
repro is here https://github.com/batzen/XAMLTools

**Comment by SimonCropp** [comment]
yes. it still returns an empty string for the exception type in eap6

**Comment by SimonCropp** [comment]
in eap7 it is now throwing a null ref

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at JetBrains.ReSharper.UnitTestFramework.Execution.UnitTestResultData.GetExceptionInfo(Int32 pos)
   at Extensions.HasVerifyException(UnitTestResultData result) in C:\Code\VerifyTests\resharper-verify\src\dotnet\ReSharperPlugin.Verify\Extensions.cs:line 11
   at ReSharperPlugin.Verify.VerifyAcceptAction.Update(IDataContext context, ActionPresentation presentation, DelegateUpdate nextUpdate) in C:\Code\VerifyTests\resharper-verify\src\dotnet\ReSharperPlugin.Verify\VerifyAcceptAction.cs:line 50
   at JetBrains.Application.UI.ActionsRevised.Handlers.ExecutableActionEvaluator.Evaluate(IAction handler, EvaluatedAction result, IDataContext context)
   at JetBrains.ReSharper.Feature.Services.Actions.ServicesActionEvaluator.Evaluate(IAction handler, EvaluatedAction result, IDataContext context)
   at JetBrains.Application.UI.ActionsRevised.Handlers.BoundAction.Evaluate(IDataContext context)
   at JetBrains.Application.UI.ActionsRevised.Handlers.ActionHandlers.Evaluate(IActionDefWithId action, IDataContext context)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActions.CreateEvaluatedActionDescriptor(IActionDefWithId actionDefWithId, IDataContext dataContext, IActionHandlers handlers)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActionsAsyncAvailability.UpdateAction(ActionUpdateRequestData req, IDataContext dataContext)
   at JetBrains.Platform.RdFramework.Actions.Backend.RdActionsAsyncAvailability.UpdateOnBackgroundThread(IPriorityQueue`1 actionsToUpdate)
```

**Comment by matkoch** [comment]
This seems to be an NUnit issue. The exception type is not available in the test run results.

**Comment by SimonCropp** [comment]
this should be fixed if u update to the current EAP of resharper/rider and 2022.1.0-eap08 of the plugin

**Comment by SimonCropp** [comment]
a stable has been released 2022.1.0
